internal/provider/services/test_helper.go

```diff
@@ -15,3 +16,8 @@ func MustMarshallString(value string, t *testing.T) types.AttributeValue {
 		return result
 	}
 }
+
+func ResolveS3Locations(providerID, providerVersion, value string) string {
```

Comment:
It will just download the provider binary for the platform from where it runs.
A prefix has been defined that locates the provider binary uniquely based on parameters such as provider name, version, os and arch.
`terraform-provider-<name>_<version>_<os>_<arch>.zip`
Ref: [Link](https://github.com/terrariumcloud/terrarium/blob/ciedev-3259-2/internal/provider/services/storage/storage.go#L77)
 These inputs we'll be passing from the endpoint that will get defined as a part of upcoming PR in providers v1 rest implementation.
`https://<host-name>/providers/v1/{organization_name}/{name}/{version}/{os}/{arch}/archive`

and terraform init is smart to retrive the metadata (if it exists) for the platform where it is running.